src/python/pants/jvm/target_types.py

```diff
@@ -309,6 +310,17 @@ class JunitTestSourceField(SingleSourceField, metaclass=ABCMeta):
 # -----------------------------------------------------------------------------------------------
 
 
+class ReproducibleJar(BoolField):
```

Comment:
Pants convention has been to name fields with a `Field` suffix. Maybe rename to `ReproducibleJarField`?